File: src/topic_tertule/src/person_publisher.cpp
/**
 * 该例程将发布/person_info话题，消息类型topic_tertule
 */

 #include <ros/ros.h>
 #include <topic_tertule/person.h>


 int main(int argc, char **argv)
 {
     // ROS节点初始化,前两个参数不变，最后为发布者名称
     ros::init(argc, argv, "person_publisher");

     // 创建节点句柄
     ros::NodeHandle n;
     
     // 创建一个publisher，发布/person_info的topic，消息类型为geometry::Twist,队列长度为10
     ros::Publisher person_info_pub = n.advertise<topic_tertule::person>("/person_info", 10);

     // 设置循环的频率
     ros::Rate loop_rate(1);

     int count = 0;
     while(ros::ok())
     {
         // 初始化topic/tertle类型的消息
         topic_tertule::person person_msg;
         person_msg.name = "WuKong";
         person_msg.age  = 1;
         person_msg.sex  = topic_tertule::person::male;

         // 发布消息
         person_info_pub.publish(person_msg);
         
         //ROS输出日志
         ROS_INFO("publish person Info: name:%s age:%d sex:%d",
                    person_msg.name.c_str(), person_msg.age, person_msg.sex);
         
         // 按照循环频率延时
         loop_rate.sleep();
     }

     return 0;

 }

File: src/topic_tertule/scripts/person_publisher.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# 该例程将发布/person_info话题，消息类型为topic_tertule::person

import rospy
from topic_tertule.msg import person

def publisher():
    # ROS节点初始化
    rospy.init_node('person_publisher', anonymous=True)

    # 创建一个publisher,发布/turtle1/cmd_vel的topic，消息类型为geometry_msgs::Twist,队列长为10
    person_info_pub = rospy.Publisher('/person_info', person, queue_size=10)

    # 设置循环频率
    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        # 初始化topic_tertle::person类型的消息
        person_msg = person()
        person_msg.name = "WuKong"
        person_msg.age  = 1
        person_msg.sex  = person.male

        # 发布消息
        person_info_pub.publish(person_msg)
        rospy.loginfo("Publish person message[%s, %d, %d]",
                        person_msg.name, person_msg.age, person_msg.sex)

        
        # 按照循环频率延时
        rate.sleep()

if __name__ == '__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass
